src/composables/pools/usePoolMigration.ts

```diff
@@ -0,0 +1,336 @@
+  });
+
+  const migrationFn = computed(() => {
+    if (migrationType.value?.type === PoolMigrationType.AAVE_BOOSTED_POOL) {
+      return migrateBoostedPool;
+    } else if (migrationType.value?.type === PoolMigrationType.STABAL3_POOL) {
+      return migrateStabal3;
+    } else {
+      return migrateBoostedPool;
+    }
+  });
+
+  const migrationData = computed(() => {
+    const signer = getSigner();
+    const signerAddress = account.value;
+    const _signature = relayerApproval.value ? undefined : signature.value;
+    const _tokens = fromPool.tokens
+      .filter(
+        token => token.address.toLowerCase() !== fromPool.address.toLowerCase()
+      )
+      .map(token =>
+        parseUnits(token.balance, fromPool.onchain?.decimals || 18).toString()
+      );
+
+    return {
+      signer,
+      signerAddress,
+      _signature,
+      _tokens,
+    };
+  });
+
+  const fiatTotal = computed((): string => {
+    if (actions.value[currentActionIndex.value].isStakeAction) {
+      return fiatValueOf(fromPool, stakedBptBalance);
+    } else if (actions.value[currentActionIndex.value].isUnstakeAction) {
+      return fiatValueOf(fromPool, unstakedBptBalance);
+    } else {
+      return '0';
+    }
+  });
+
+  const fiatTotalLabel = computed((): string => {
+    return fNum2(fiatTotal.value, FNumFormats.fiat);
+  });
+
+  /**
+   * METHODS
+   */
+  async function getUserSignature(): Promise<string> {
+    try {
+      const deadline = new Date().getTime() + oneHourInMs / 2;
+      const vaultInstance = balancerContractsService.vault?.instance;
+      const nonce = await vaultInstance.getNextNonce(account.value);
+      const relayerAddress = configService.network.addresses.batchRelayer;
+      const calldata = vaultInstance.interface.encodeFunctionData(
+        'setRelayerApproval',
+        [account.value, relayerAddress, true]
+      );
+
+      const _signature =
+        await RelayerAuthorization.signSetRelayerApprovalAuthorization(
+          vaultInstance,
+          getSigner(),
+          relayerAddress,
+          calldata,
+          deadline,
+          nonce
+        );
+
+      const encodedSign = RelayerAuthorization.encodeCalldataAuthorization(
+        '0x',
+        deadline,
+        _signature
+      );
+
+      signature.value = encodedSign;
+      return signature.value;
+    } catch (error) {
+      console.error(error);
+      throw error;
+    }
+  }
+
+  async function migrate(
+    amount: string,
+    isStaked: boolean
+  ): Promise<TransactionResponse> {
+    let query = migrationFn.value(amount, isStaked, '0');
+
+    const estimatedGas = await getSigner().estimateGas({
+      to: query.to,
+      data: query.data,
+    });
+
+    const gasLimit = Math.floor(
+      BigNumber.from(estimatedGas).toNumber() * (1 + GAS_LIMIT_BUFFER)
+    );
+
+    const expectedBptOut = await getExpectedBptOut(amount, isStaked);
+    query = migrationFn.value(amount, isStaked, expectedBptOut); // TODO apply slippage here.
```

Comment:
For when you get back @alter-eggo, the `expectedBptOut` here should not be passed directly as a limit. This 'limit' value is actually the minBptOut which is the minimum amount of BPT the user wants to receive or the transaction should fail. This is why we have a user setting called slippage. Please use it here to adjust this value, see `useSlippage` composable. You probably want to do something like:
```suggestion
    const minBptOut = minusSlippageScaled(expectedBptOut);
    query = migrationFn.value(amount, isStaked, minBptOut); // TODO apply slippage here.
```